### PR TITLE
add optimize SLI hash method

### PR DESCRIPTION
**Pull request by sadam21**

### CODE REVIEW

File: src/main/java/com/gooddata/dataset/DatasetService.java [hunk 1]
```diff
@@ -196,4 +199,44 @@ private String getErrorMessage(final PullTaskStatus status, final Path dirPath)
             throw new GoodDataException("Unable to list datasets for project " + project.getId(), e);
         }
     }
+
+    /**
+     * Optomize SLI hash. This feature is useful only if data warehouse was reduced somehow. Remove unused values from
```
Comment (martiner) [inline src/main/java/com/gooddata/dataset/DatasetService.java]: typo

File: src/main/java/com/gooddata/dataset/DatasetService.java [hunk 2]
```diff
@@ -196,4 +199,44 @@ private String getErrorMessage(final PullTaskStatus status, final Path dirPath)
...
+            @Override
+            public boolean isFinished(final ClientHttpResponse response) throws IOException {
+                if (!super.isFinished(response)) {
+                    return false;
+                }
+                final MaqlDdlTaskStatus maqlDdlTaskStatus = extractData(response, MaqlDdlTaskStatus.class);
+                if (maqlDdlTaskStatus.isSuccess()) {
+                    return true;
+                }
+                throw new GoodDataException("Unable to update project model: " + maqlDdlTaskStatus.getMessages());
```
Comment (martiner) [inline src/main/java/com/gooddata/dataset/DatasetService.java]: project model? ;)

File: src/main/java/com/gooddata/dataset/EtlMode.java [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+import org.codehaus.jackson.annotate.JsonProperty;
+import org.codehaus.jackson.annotate.JsonTypeInfo;
+import org.codehaus.jackson.annotate.JsonTypeName;
+import org.codehaus.jackson.map.annotate.JsonSerialize;
+
+@JsonTypeName("etlMode")
+@JsonTypeInfo(include = JsonTypeInfo.As.WRAPPER_OBJECT, use = JsonTypeInfo.Id.NAME)
+@JsonIgnoreProperties(ignoreUnknown = true)
+@JsonSerialize(include = JsonSerialize.Inclusion.NON_NULL)
+public class EtlMode {
```
Comment (martiner) [inline src/main/java/com/gooddata/dataset/EtlMode.java]: package protected

File: src/main/java/com/gooddata/dataset/EtlModeType.java [hunk 4]
```diff
@@ -0,0 +1,16 @@
+package com.gooddata.dataset;
+
+import org.codehaus.jackson.annotate.JsonValue;
+
+/**
+ * Enum containing ETL mode types.
+ */
+public enum EtlModeType {
```
Comment (martiner) [inline src/main/java/com/gooddata/dataset/EtlModeType.java]: package protected

File: src/main/java/com/gooddata/dataset/LookupMode.java [hunk 5]
```diff
@@ -0,0 +1,16 @@
+package com.gooddata.dataset;
+
+import org.codehaus.jackson.annotate.JsonValue;
+
+/**
+ * Enum containing ETL lookup modes.
+ */
+public enum LookupMode {
```
Comment (martiner) [inline src/main/java/com/gooddata/dataset/LookupMode.java]: package protected

File: src/main/java/com/gooddata/dataset/DatasetService.java (hunk 1 above)
Comment (sadam21) [inline src/main/java/com/gooddata/dataset/DatasetService.java]: Corrected.

File: src/main/java/com/gooddata/dataset/DatasetService.java (hunk 2 above)
Comment (sadam21) [inline src/main/java/com/gooddata/dataset/DatasetService.java]: Error message corrected :)

File: src/main/java/com/gooddata/dataset/EtlMode.java (hunk 3 above)
Comment (sadam21) [inline src/main/java/com/gooddata/dataset/EtlMode.java]: Done

File: src/main/java/com/gooddata/dataset/EtlModeType.java (hunk 4 above)
Comment (sadam21) [inline src/main/java/com/gooddata/dataset/EtlModeType.java]: Done

File: src/main/java/com/gooddata/dataset/LookupMode.java (hunk 5 above)
Comment (sadam21) [inline src/main/java/com/gooddata/dataset/LookupMode.java]: Done

**Comment by martiner** [comment]
close #110

**Pull request merged**